components/automate-cli/cmd/chef-automate/cleanup.go

```diff
@@ -0,0 +1,172 @@
+		sudo rm -rf /hab;
+		`
+	
+		BACKENDCLEANUP_COMMANDS = `
+		sudo systemctl stop hab-sup;
+		sudo rm -rf /hab;   
+		`
+)
+
+func runCleanupCmd(cmd *cobra.Command, args []string) error {
+	infra, err := getAutomateHAInfraDetails()
+	if err != nil {
+		return err
+	}
+	if infra != nil {
+		sshInfo :=SshClient{
+			sshUser: infra.Outputs.SSHUser.Value,
+			sshPort: infra.Outputs.SSHPort.Value,
+			sshKeyFile: infra.Outputs.SSHKeyFile.Value,
+		}
+
+			writer.Printf(strings.Join(args, ""))
+		if isA2HARBFileExist() {
+			if cleanupFlags.onprem {
+				automateIps := infra.Outputs.AutomatePrivateIps.Value
+				chefserverIps := infra.Outputs.AutomatePrivateIps.Value
+				postgresqlIps := infra.Outputs.PostgresqlPrivateIps.Value
+				opensearchIps := infra.Outputs.OpensearchPrivateIps.Value
+				for i := 0; i < len(automateIps); i++ {
+					servername := "Automate"
+					err := executeCleanupOnRemote(sshInfo, automateIps[i], servername, FRONTENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				for i := 0; i < len(chefserverIps); i++ {
+					servername := "chef server"
+					err := executeCleanupOnRemote(sshInfo, chefserverIps[i], servername, FRONTENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				for i := 0; i < len(postgresqlIps); i++ {
+					servername := "postgresql"
+					err := executeCleanupOnRemote(sshInfo, postgresqlIps[i], servername, BACKENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				for i := 0; i < len(opensearchIps); i++ {
+					servername := "opensearch"
+					err := executeCleanupOnRemote(sshInfo, opensearchIps[i], servername, BACKENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				cleanUpScript := "hab pkg uninstall chef/automate-ha-deployment"
+				args := []string{
+					"-c",
+					cleanUpScript,
+				}
+				err := executeCommand("/bin/sh", args, "")
+				if err != nil {
+					return err
+				}
+				return err
+			}
+		}
+	} else {
+		fmt.Println("\nCleanup cannot be done")
+	}
+	return err
+}
+
+func executeCleanupOnRemote(sshDetails SshClient, ip string, servername string, commands string) error {
+	// read private key file
+	pemBytes, err := ioutil.ReadFile(sshDetails.sshKeyFile)
+	if err != nil {
+		writer.Errorf("Unable to read private key: %v", err)
+		return err 
+		}
+		// create signer
+	signer, err := ssh.ParsePrivateKey(pemBytes)
+	if err != nil {
+		fmt.Printf("Parsing key failed: %v", err)
+		os.Exit(1)
+		// writer.Errorf("Parsing key failed: %v", err)
+		// return err
+	}
+	config := &ssh.ClientConfig{
+		User: sshDetails.sshUser,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+		HostKeyCallback: func(hostname string, remote net.Addr, key ssh.PublicKey) error {
+			// use OpenSSH's known_hosts file if you care about host validation
```

Comment:
Host validation is mandatory in production. Change the callback function to add the public key in known_hosts

Reply:
done